google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -1950,7 +1957,17 @@ <ReqT, RespT> GrpcCallContext newCallContext(
       boolean routeToLeader) {
     GrpcCallContext context = GrpcCallContext.createDefault();
     if (options != null) {
+      // Set channel affinity in GAX. This is a no-op for gRPC-GCP.
       context = context.withChannelAffinity(Option.CHANNEL_HINT.getLong(options).intValue());
+
+      // Set channel affinity in gRPC-GCP. This is a no-op for GAX.
+      context =
+          context.withCallOptions(
+              context
+                  .getCallOptions()
+                  .withOption(
+                      GcpManagedChannel.AFFINITY_KEY,
+                      Option.CHANNEL_HINT.getLong(options).toString()));
```

Comment:
This channel hint can be any random Long, meaning that we might be using a very large range of different channel hints. That again means that we might be adding an unbounded set of channel hints to the map in grpc-gcp. We should add a computation here that makes sure that the set of hints is limited (e.g. do something like `hint % options.getNumChannels()`).

See https://github.com/googleapis/java-spanner/blob/e859b29ccf4e68b1ab62cffdd4cf197011ba9878/google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractReadContext.java#L342